compiler/destroyer.nim

```diff
@@ -395,6 +365,13 @@ proc destructiveMoveVar(n: PNode; c: var Con): PNode =
   result.add genWasMoved(n, c)
   result.add tempAsNode
 
+proc sinkParamIsLastReadCheck(c: var Con, s: PNode) = 
+  assert s.kind == nkSym and s.sym.kind == skParam
+  discard isLastRead(s, c)
+  if c.otherRead != nil:
```

Comment:
It is unclear to me why ``c.otherRead`` is used instead of ``isLastRead``'s result.